data/datasets/base.py

```diff
@@ -23,6 +23,10 @@ def __init__(
         super().__init__()
         self.root = root
         self.split = split
```

Comment:
hey that's dynamic splitting - really cool. If we make the split dynamically then maybe also lets make those parameters settable via __init__() (add them as kwargs with default values)